lib/IRGen/GenObjC.cpp

```diff
@@ -1266,8 +1303,13 @@ irgen::emitObjCSetterDescriptorParts(IRGenModule &IGM,
   descriptor.selectorRef = IGM.getAddrOfObjCMethodName(setterSel.str());
   auto methodTy = getObjCMethodType(IGM,
                               subscript->getOpaqueAccessor(AccessorKind::Set));
-  descriptor.typeEncoding = getObjCEncodingForMethodType(IGM, methodTy,
-                                                         /*extended*/false);
+  if (auto e =
+          getObjectEncodingFromClangNode(IGM, subscript, false /*extended*/)) {
+    descriptor.typeEncoding = e;
+  } else {
+    descriptor.typeEncoding = getObjCEncodingForMethodType(IGM, methodTy,
+                                                           /*extended*/ false);
+  }
```

Comment:
Rather than having the same `if`/`else` construct in four places, can we teach `getObjCEncodingForMethodType` to take a `Decl *` and, if it's non-null and has a Clang node, use Clang?

Reply:
Sure. https://github.com/apple/swift/pull/30449